src/Psalm/Internal/Analyzer/Statements/UnusedAssignmentRemover.php

```diff
@@ -45,7 +47,28 @@ public function findUnusedAssignment(
         [$assign_stmt, $assign_exp] = $search_result;
         $chain_assignment = false;
 
-        if ($assign_stmt !== null && $assign_exp !== null) {
+        if ($assign_stmt instanceof PhpParser\Node\Stmt\Catch_) {
+            assert($assign_stmt->var !== null && is_string($assign_stmt->var->name));
+            if ($codebase->analysis_php_version_id < 8_00_00) {
+                $new_file_manipulation = new FileManipulation(
+                    $original_location->raw_file_start + 1,
+                    $original_location->raw_file_start + 1 + strlen($assign_stmt->var->name),
+                    "_",
+                );
+            } else {
+                $file_content = $codebase->file_provider->getContents(
+                    $original_location->file_path
+                );
+                $end_of_exception = $original_location->raw_file_start - 1;
+                for (; ctype_space($file_content[$end_of_exception]); --$end_of_exception);
```

Comment:
this is to handle multiple exception on a single catch, right?

Reply:
That's to remove all whitespace. `catch (Exception|Error $e)` becomes `catch (Exception|Error)`, but also
```php
catch (Exception|Error
    $e)
```
becomes
```php
catch (Exception|Error)
```
instead of
```php
catch (Exception|Error
    )
```
I imagine it's not too relevant most of the time, but maybe there are multiple lines for some very long exception names. I also noticed a couple of places where we have 2 spaces between the exception and the variable, so it fixes that as well.